src/crimson/osd/main_config_bootstrap_helpers.cc

```diff
@@ -84,6 +84,54 @@ seastar::future<> populate_config_from_mon()
   });
 }
 
+struct first_double {
+  double value;
+  first_double() : value(0.0) {}
+  first_double(double v) : value(v) {}
+};
+struct second_double {
+  double value;
+  second_double() : value(0.0) {}
+  second_double(double v) : value(v) {}
+};
+
+using OptionValue = std::variant<first_double, second_double, uint64_t>;
+struct SeastarOption {
+  std::string option_name;          // Command-line option name
+  std::string config_key;           // Configuration key
+  OptionValue value_type;           // Configuration value type
+};
+
+// Define a map of options with their corresponding configuration keys and value type
+// In future to add an option introduce a new entry in this map with corresponding key name and value type
+// In case if same value type exists in the map then declare a new struct like first_double and second_double
+const std::vector<SeastarOption> seastar_options = {
+  {"--task-quota-ms", "crimson_reactor_task_quota_ms", first_double{}},
+  {"--io-latency-goal-ms", "crimson_reactor_io_latency_goal_ms", second_double{}},
+  {"--idle-poll-time-us", "crimson_reactor_idle_poll_time_us", uint64_t()}};
+
+std::optional<OptionValue> get_option_value(const SeastarOption& option) {
```

Comment:
Can we return a string here instead of `OptionValue` and drop `OptionValue`?

Reply:
We have to convert it then we can return